composer/loggers/neptune_logger.py

```diff
@@ -0,0 +1,307 @@
+            ``NEPTUNE_PROJECT`` environment variable will be used.
+        api_token (str, optional): Your Neptune API token.
+            You can leave out this argument if you save your token to the
+            ``NEPTUNE_API_TOKEN`` environment variable (recommended).
+            You can find your API token in the user menu of the Neptune web app.
+        rank_zero_only (bool, optional): Whether to log only on the rank-zero process.
+            (default: ``True``).
+        upload_artifacts (bool, optional): Whether the logger should upload artifacts to Neptune.
+            (default: ``False``).
+        base_namespace (str, optional): The name of the base namespace to log the metadata to.
+            (default: "training").
+        neptune_kwargs (Dict[str, Any], optional): Any additional keyword arguments to the
+            ``neptune.init_run()`` function. For options, see the
+            `Run API reference <https://docs.neptune.ai/api/neptune/#init_run>`_ in the
+            Neptune docs.
+    """
+    metric_namespace = 'metrics'
+    hyperparam_namespace = 'hyperparameters'
+    trace_namespace = 'traces'
+    integration_version_key = 'source_code/integrations/neptune-MosaicML'
+
+    def __init__(
+        self,
+        *,
+        project: Optional[str] = None,
+        api_token: Optional[str] = None,
+        rank_zero_only: bool = True,
+        upload_artifacts: bool = False,
+        base_namespace: str = 'training',
+        **neptune_kwargs,
+    ) -> None:
+        try:
+            from neptune.internal.utils import verify_type
+        except ImportError as e:
+            raise MissingConditionalImportError(extra_deps_group='neptune',
+                                                conda_package='neptune',
+                                                conda_channel='conda-forge') from e
+
+        verify_type('project', project, (str, type(None)))
+        verify_type('api_token', api_token, (str, type(None)))
+        verify_type('rank_zero_only', rank_zero_only, bool)
+        verify_type('upload_artifacts', upload_artifacts, bool)
+        verify_type('base_namespace', base_namespace, str)
+
+        if not base_namespace:
+            raise ValueError("Argument 'base_namespace' cannot be an empty string.")
+
+        self._project = project
+        self._api_token = api_token
+        self._rank_zero_only = rank_zero_only
+        self._upload_artifacts = upload_artifacts
+        self._base_namespace = base_namespace
+        self._neptune_kwargs = neptune_kwargs
+
+        mode = self._neptune_kwargs.pop('mode', 'async')
+
+        self._enabled = (not rank_zero_only) or dist.get_global_rank() == 0
+
+        self._mode = mode if self._enabled else 'debug'
+
+        self._neptune_run = None
+        self._base_handler = None
+
+        self._metrics_dict: Dict[str, int] = {}  # used to prevent duplicate step logging
+
+        super(NeptuneLogger, self).__init__()
```

Comment:
This doesn't need a super init because the base class is ABC without an init, but if you want to leave it, prefer the syntax `super().__init__()`

Reply:
I just wanted PyCharm to stop complaining about lack of call to the super class init :laughing:

Reply:
That's fine. I'll edit and we can merge.